data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,562 @@
+```ocaml
+# print_endline "ha ha";;
+ha ha
+- : unit = ()
+
+# let _ = print_endline  "ha ha";;
+ha ha
+- : unit = ()
+```
+
+As seen in the last example, the catch-all pattern (the underscore symbol) can be used in definitions. The following example illustrates its use, which is distinct from the unit pattern:
+```ocaml
+# let (_, y) = List.split [(1,2); (3,4); (5,6); (7,8)];;
+val y : int list = [2; 4; 6; 8]
+```
+
+### Local Definitions
+
+Local definitions are like global definitions, except the name is only bound inside an expression. They are introduced by the `let ... = ... in ...` construction. The name bound before the `in` keyword is only bound in the expression after the `in` keyword
+```ocaml
+# let b = 2 * 3 in b * 7;;
+- : int = 42
+
+# b;;
+Error: Unbound value b
+```
+
+Here, the name `b` is bound to `6` inside the expression `b * 7`. A couple of remarks:
+- No global definition is introduced in this example
+- Computation of `2 * 3` will always take place before `b * 7`
+
+Local definitions can be chained (one after another) or nested (one inside another). Here is an example of chaining:
+```ocaml
+# let b = 2 * 3 in
+  let c = b * 7 in
+  b * c;;
+- : int = 252
+```
+
+This is how scoping works here:
+- `b` is bound to `6` inside `let c = b * 7 in b * c`
+- `c` is bound to `42` inside `b * c`
+
+Here is an example of nesting:
+```ocaml
+# let b =
+    let c = 2 * 3 in
+    c * 5 in
+  b * 7;;
+- : int = 210
+Here is how scoping works:
+- `c` is bound to `6` inside `c * 5`
+- `b` is bound to `30` inside `b * 7`
+
+Arbitrary combinations or chaining and nesting are allowed.
+
+### Scopes and Environments
+
+Without oversimplifying, an OCaml program is a sequence of definitions or expressions. These items are said to be at the _toplevel_. That's why an OCaml REPL, such as UTop, is called a toplevel, because that's where typed expressions go.
+
+Execution evaluates each expression from top to bottom.
+
+At any time during evaluation, the _environment_ is the ordered sequence of available definitions.
+
+Here, the name `tau` is added into the toplevel environment.
+```ocaml
+# let tau = 6.28318;;
+val tau : float = 6.28318
+```
+
+The scope of `tau` is global. This name is available anywhere after the definition.
+
+Here, the global environment is unchanged:
+```ocaml
+# let pi = 3.14159 in 2. *. pi;;
+- : float = 6.28318
+```
+
+However, with respect to the expression `2. *. pi`, the environment is different, as it contains the definition of `pi`. Local definition create local environments.
+
+The scope of a definition is the set of environments where it is reachable.
+
+Although OCaml is an expression-oriented language, it is not entirely free of statements. The `let` construct is a statement. It is a statement that may change the state of the environment by adding a name-value binding. Note these expressions at the toplevel also fall into that category because they are equivalent to `let _ =` definitions.
```

Comment:
```suggestion
Although OCaml is an expression-oriented language, it is not entirely free of statements. The `let` construct is a statement that may change the state of the environment by adding a name-value binding. Note these expressions at the toplevel also fall into that category because they are equivalent to `let _ =` definitions.
```

Would it be better to say it this way without the repetition of "is a statement," or is that repetition there for emphasis?